spec/hydra/pcdm/models/object_spec.rb

```diff
@@ -2,28 +2,228 @@
 
-  describe '#objects=' do
-    it 'should aggregate objects' do
-      object1.objects = [object2, object3]
-      object1.save
-      expect(object1.objects).to eq [object2, object3]
+      it 'should add objects objects' do
+        subject.child_objects = [object1,object2]
+        subject.child_objects += [object3]
+        subject.child_objects += [object4,object5]
+        expect( subject.child_objects ).to eq [object1,object2,object3,object4,object5]
+     end
+
+      it 'should allow sub-objects' do
+        subject.child_objects = [object1,object2]
+        object1.child_objects = [object3]
+        expect( subject.child_objects ).to eq [object1,object2]
+        expect( object1.child_objects ).to eq [object3]
+      end
+
+      it 'should allow repeating objects' do
+      skip( "pending resolution of PCDM-105 and AF-853") do
+        subject.child_objects = [object1,object2]
+        subject.child_objects += [object1]
+        expect( subject.child_objects ).to eq [object1,object2,object1]
+      end
+      end
+
+      describe 'adding objects that are ancestors' do
+        let(:error_type)    { ArgumentError }
+        let(:error_message) { "an object can't be an ancestor of itself" }
+
+        context 'when the source object is the same' do
+          it 'raises an error' do
+            expect{ object1.child_objects = [object1] }.to raise_error(error_type, error_message)
+            expect{ object1.child_objects += [object1] }.to raise_error(error_type, error_message)
+          end
+        end
+
+        before do
+          object1.child_objects = [object2]
+        end
+
+        it 'raises an error' do
+          expect{ object2.child_objects += [object1] }.to raise_error(error_type, error_message)
+          expect{ object2.child_objects = [object1] }.to raise_error(error_type, error_message)
+        end
+
+        context 'with more ancestors' do
+          before do
+            object2.child_objects = [object3]
+          end
+
+          it 'raises an error' do
+            expect{ object3.child_objects = [object1] }.to raise_error(error_type, error_message)
+            expect{ object3.child_objects += [object1] }.to raise_error(error_type, error_message)
+          end
+
+          context 'with a more complicated example' do
+            before do
+              object3.child_objects = [object4,object5]
+            end
+
+            it 'raises errors' do
+              expect{ object4.child_objects = [object1] }.to raise_error(error_type, error_message)
+              expect{ object4.child_objects += [object1] }.to raise_error(error_type, error_message)
+
+              expect{ object4.child_objects = [object2] }.to raise_error(error_type, error_message)
+              expect{ object4.child_objects += [object2] }.to raise_error(error_type, error_message)
+            end
+          end
+        end
+      end
+    end
+
+    context 'with unacceptable child objects' do
```

Comment:
All these tests are hitting the same code path in Association#raise_if_invalid. Can we test all the variations of that method in a unit test rather than an integration test?